### PR TITLE
Typings for Firefox Addon SDK/jpm

### PR DESCRIPTION
**Pull request by realmhamdy**
case 1. Add a new type definition.
- [ ] checked compilation succeeds with `--target es6` and `--noImplicitAny` options.
- [ ] has correct [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)
- [ ] has a [test file](http://definitelytyped.org/guides/contributing.html#tests) with the suffix of  `-tests.ts` or `-tests.tsx`.

This has also been requested in this [issue](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/4667)

### CODE REVIEW

File: jpm/jpm.d.ts [hunk 1]
```diff
@@ -0,0 +1,1114 @@
+// Type definitions for Firefox Addon SDK
+// Project: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/Add-on_SDK
+// Definitions by: Mohammed Hamdy <https://github.com/github-account-because-they-want-it>
```
Comment (mhegazy) [inline jpm/jpm.d.ts]: can you update the link.

Comment (mhegazy) [inline jpm/jpm.d.ts]: The CI uses this to ping authors to review any changes in the definition later on.

**Comment by mhegazy** [comment]
Thank you for your contribution. 

Can you please fix the failing Travis CI tests and update the PR!

File: jpm/jpm.d.ts (hunk 1 above)
Comment (mhegazy) [inline jpm/jpm.d.ts]: never mind, that is your real account :) sorry about that.

**Comment by github-account-because-they-want-it** [comment]
@mhegazy All green.

**Comment by github-account-because-they-want-it** [comment]
@mhegazy you can also close #4667 while still at it.

**Pull request merged**